10up/classifai
# Feature: Azure PDF scan

### Description of the Change
This PR utilizes the Azure Computer Vision Read API to extract text from multi-page PDF files. It supports both textbase and text-heavy image base pdf files.

Because of the Read API design, this feature uses WP Cron to periodically check and grab the result.



### Verification Process

1. Go to ClassifAI > Image Processing.
2. See the new setting `Enable Scanning PDF`.
3. Enable that feature.
4. Upload a PDF file.
5. Right after the file is uploaded, open its media modal, see `Classifai Read PDF` field with a disabled `In progress!` button.
6. Wait for some minutes for API to process the file, check the modal again, see the description field filled with content of the PDF file.
7. Open the attachment detail page, see a new metabox `Classifai PDF Processing` with rescan checkbox.



### Checklist:






### Applicable Issues



### Changelog Entry
